app/helpers/file_helper.rb

```diff
@@ -18,8 +18,8 @@ def accept_file(file, name, kind)
       accept = ['image/png', 'image/gif', 'image/bmp', 'image/tiff', 'image/jpeg', 'image/x-ms-bmp']
     when 'code'
       accept = ['text/x-pascal', 'text/x-c', 'text/x-c++', 'text/plain', 'text/', 'application/javascript, text/html',
-                'text/css', 'text/x-ruby', 'text/coffeescript', 'text/x-scss', 'application/json', 'text/xml', 'application/xml',
-                'text/x-yaml', 'application/xml', 'text/x-typescript']
+        'text/css', 'text/x-ruby', 'text/coffeescript', 'text/x-scss', 'application/json', 'text/xml', 'application/xml',
+        'text/x-yaml', 'application/xml', 'text/x-typescript','text/x-hdl','text/x-asm','text/x-jack','text/x-hack']
```

Comment:
Are these the right mime types that will be identified for these file types? Maybe have a look for some "Elements of Computer Systems" hdl, asm, jack and hack file samples. There are some github repos with these in.